capstone/src/main/webapp/scripts/post.js

```diff
@@ -0,0 +1,193 @@
+function createAuthor(post) {
+  const postAuthor = document.createElement('h3');
+  postAuthor.className = "post-author align-vertical";
+  postAuthor.innerText = post.authorId;
+  return postAuthor;
+}
+
+// Create HTML element for post text
+function createPostText(post) {
+  const postContent = document.createElement('p');
+  postContent.className = "post-content";
+  postContent.innerText = post.postText;
+  return postContent;
+}
+
+function createLikesContainer(post) {
+  const likesDiv = document.createElement('div');
+  likesDiv.className = "likes-div";
+
+  const likesLabel = document.createElement('p');
+  likesLabel.className = "likes-label vertical-align";
+  likesLabel.id = post.postId + "likes-label";
+  likesLabel.innerText = "0 likes";
+  likesDiv.appendChild(likesLabel);
+
+  const likeIcon = document.createElement('ion-icon');
+  likeIcon.className = "like-icon vertical-align";
+  likeIcon.name = "heart";
+  likeIcon.id = post.postId + "like";
+  likesDiv.appendChild(likeIcon);
+
+  return likesDiv;
+}
+
+// Create container for all post comments
+function createCommentsContainer(post) {
+  const commentsContainer = document.createElement('div');
+  commentsContainer.className = "comments-content";
+  const allComments = document.createElement('ul');
+  for(comment of post.comments) {
+    allComments.appendChild(createSingleComment(comment));
+  }
+  commentsContainer.appendChild(allComments);
+  return commentsContainer;
+}
+
+function createSingleComment(comment) {
+  const commentContainer = document.createElement('li');
+  commentContainer.className = "comment-content";
+
+  const commentUserImg = document.createElement('span');
+  commentUserImg.className = "comment-user align-vertical";
+  commentContainer.appendChild(commentUserImg);
+
+  const commentTextDiv = document.createElement('div');
+  commentTextDiv.className = "comment-text-div align-vertical";
+  commentContainer.appendChild(commentTextDiv);
+
+  const commentText = document.createElement('p');
+  commentText.className = "comment-text align-vertical";
+  commentText.innerText = comment.commentText;
+  commentTextDiv.appendChild(commentText);
+
+  return commentContainer;
+}
+
+// Create comment input HTML element
+function createCommentBox(post) {
+  const commentBoxDiv = document.createElement('div');
+  commentBoxDiv.className = "comment-box-div";
+
+  const commentBox = document.createElement('input');
+  commentBox.type = "text";
+  commentBox.name = "comment-input";
+  commentBox.placeholder = "Write a comment";
+  commentBox.className = "comment-input";
+  commentBox.id = post.postId + "comment-input";
+  commentBoxDiv.appendChild(commentBox);
+
+  const commentBtn = document.createElement('button');
+  commentBtn.className = "post-btn align-vertical comment-btn";
+  commentBtn.type = "submit";
+  commentBtn.id = post.postId;
+  commentBtn.innerHTML = "<img class='small-icon' src='images/send_plane.png' alt/>";
+  commentBtn.onclick = "postComment()";
```

Comment:
Doesn't postComment() take parameters? Does this still work?

Reply:
forgot to remove